src/hyperion/experiment_plans/oav_grid_detection_plan.py

```diff
@@ -113,13 +93,17 @@ def grid_detection_main_plan(
         # need to wait for the OAV image to update
         # See #673 for improvements
         yield from bps.sleep(OAV_REFRESH_DELAY)
-
-        tip_x_px, tip_y_px = yield from wait_for_tip_to_be_found(oav.mxsc.pin_tip)
+        tip_x_px, tip_y_px = yield from trigger_and_return_pin_tip(pin_tip_detection)
+        if tip_x_px is None or tip_y_px is None:
+            timeout = yield from bps.rd(pin_tip_detection.validity_timeout)
+            raise WarningException(f"No pin found after {timeout} seconds")
```

Comment:
Should: Isn't this exactly what `wait_for_tip_to_be_found` does? Why not use that here?